letsencrypt/client/tests/revoker_test.py

```
@@ -0,0 +1,390 @@
+
+    def test_no_installer_get_installed_locations(self):
+        # pylint: disable=protected-access
+        revoker = self._get_revoker(None)
+        self.assertEqual(revoker._get_installed_locations(), {})
+
+    def test_get_installed_locations(self):
+        # pylint: disable=protected-access
+        mock_installer = mock.MagicMock()
+        mock_installer.get_all_certs_keys.return_value = self.certs_keys
+
+        revoker = self._get_revoker(mock_installer)
+        sha_vh = revoker._get_installed_locations()
+
+        self.assertEqual(len(sha_vh), 2)
+        for i, cert in enumerate(self.certs):
+            self.assertTrue(cert.get_fingerprint() in sha_vh)
+            self.assertEqual(
+                sha_vh[cert.get_fingerprint()], self.installs[i])
+
+    @mock.patch("letsencrypt.client.revoker.M2Crypto.X509.load_cert")
+    def test_get_installed_load_failure(self, mock_m2):
+        mock_installer = mock.MagicMock()
+        mock_installer.get_all_certs_keys.return_value = self.certs_keys
+
+        mock_m2.side_effect = IOError
+
+        revoker = self._get_revoker(mock_installer)
+
+        # pylint: disable=protected-access
+        self.assertEqual(revoker._get_installed_locations(), {})
+
+
+class RevokerClassMethodsTest(RevokerBase):
+    def setUp(self):
+        super(RevokerClassMethodsTest, self).setUp()
+        self.mock_config = mock.MagicMock(cert_key_backup=self.backup_dir)
+
+    def tearDown(self):
+        shutil.rmtree(self.backup_dir)
+
+    def _call(self, cert_path, key_path):
+        from letsencrypt.client.revoker import Revoker
+        Revoker.store_cert_key(cert_path, key_path, self.mock_config)
+
+    def test_store_two(self):
+        from letsencrypt.client.revoker import Revoker
+        self._call(self.paths[0], self.key_path)
+        self._call(self.paths[1], self.key_path)
+
+        self.assertTrue(os.path.isfile(self.list_path))
+        rows = self._get_rows()
+        i = 0
```

If there are not any rows then won't 'i' be undefined for line 300.  I guess since it is a test it will still fail, but it seems like you would rather it fail the assert then have an undefined variable.